### ISSUE TITLE
Beta 2.5.22: PLEX.music: Item not yet synced

### ISSUE THREAD

**Issue by dazedcrazy** [croneter/PlexKodiConnect#622]
```12:44:22.926 T:7372  NOTICE: PLEX.music: Item not yet synced: {'art': '/library/metadata/52839/art/1548270938', 'parentTitle': 'Queen', 'addedAt': '566438400', 'parentThumb': '/library/metadata/52839/thumb/1548270938', 'ratingKey': '56191', 'deepAnalysisVersion': '2', 'key': '/library/metadata/56191/children', 'year': '1987', 'updatedAt': '1548270865', 'originallyAvailableAt': '1987-01-23', 'index': '1', 'titleSort': 'Day at the Races', 'thumb': '/library/metadata/56191/thumb/1548270865', 'title': 'A Day at the Races', 'type': 'album', 'parentRatingKey': '52839', 'summary': '', 'parentKey': '/library/metadata/52839'}```
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/2798978/kodi.log)
Then when i reset Kodi i get a different item not synced
```12:59:09.465 T:10200  NOTICE: PLEX.music: Item not yet synced: {'parentThumb': '/library/metadata/52839/thumb/1548270938', 'art': '/library/metadata/52839/art/1548270938', 'viewCount': '2', 'parentTitle': 'Queen', 'addedAt': '1303167600', 'updatedAt': '1548270877', 'ratingKey': '52840', 'deepAnalysisVersion': '2', 'key': '/library/metadata/52840/children', 'year': '2011', 'originallyAvailableAt': '2011-04-19', 'index': '1', 'lastViewedAt': '1544103343', 'thumb': '/library/metadata/52840/thumb/1548270877', 'title': 'Greatest Hits II', 'parentKey': '/library/metadata/52839', 'parentRatingKey': '52839', 'summary': '', 'type': 'album'}```

**Comment by croneter** [comment]
But then that item gets synced exactly once, correct?

**Comment by dazedcrazy** [comment]
There is no ``Get Info`` like there is with video files, how do i get it?

**Comment by croneter** [comment]
Does this always concern the same 2 items when you reset the Kodi database?

**Comment by dazedcrazy** [comment]
Yeah always the same 2 items

**Comment by dazedcrazy** [comment]
I reset the db again to check and restarted Kodi after a full sync and now i see this error
```xml
19:36:52.185 T:3414160176   ERROR: PLEX.utils: _____________________________________________________________
19:36:52.186 T:3414160176   ERROR: PLEX.utils:     Traceback (most recent call last):
19:36:52.186 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 173, in run
19:36:52.188 T:3414160176   ERROR: PLEX.utils:         self._run_internal()
19:36:52.188 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 253, in _run_internal
19:36:52.189 T:3414160176   ERROR: PLEX.utils:         self.start_library_sync(block=True)
19:36:52.189 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 128, in start_library_sync
19:36:52.190 T:3414160176   ERROR: PLEX.utils:         library_sync.start(show_dialog, repair, self.on_library_scan_finished)
19:36:52.190 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/full_sync.py", line 426, in start
19:36:52.191 T:3414160176   ERROR: PLEX.utils:         FullSync(repair, callback, show_dialog).run()
19:36:52.192 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/utils.py", line 1075, in wrapper
19:36:52.192 T:3414160176   ERROR: PLEX.utils:         result = func(*args, **kwargs)
19:36:52.193 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/full_sync.py", line 403, in run
19:36:52.194 T:3414160176   ERROR: PLEX.utils:         if not self.full_library_sync():
19:36:52.194 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/full_sync.py", line 380, in full_library_sync
19:36:52.195 T:3414160176   ERROR: PLEX.utils:         ctx.remove(plex_id, plex_type)
19:36:52.196 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/itemtypes/movies.py", line 246, in remove
19:36:52.196 T:3414160176   ERROR: PLEX.utils:         self.kodidb.delete_artwork(kodi_id, kodi_type)
19:36:52.197 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/kodi_db/common.py", line 167, in delete_artwork
19:36:52.197 T:3414160176   ERROR: PLEX.utils:         self.delete_cached_artwork(row[0])
19:36:52.198 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/kodi_db/common.py", line 34, in wrapper
19:36:52.199 T:3414160176   ERROR: PLEX.utils:         return method(self, *args, **kwargs)
19:36:52.199 T:3414160176   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/kodi_db/common.py", line 172, in delete_cached_artwork
19:36:52.200 T:3414160176   ERROR: PLEX.utils:         self.artcursor.execute("SELECT cachedurl FROM texture WHERE url = ? LIMIT 1",
19:36:52.200 T:3414160176   ERROR: PLEX.utils:     AttributeError: 'NoneType' object has no attribute 'execute'
19:36:52.201 T:3414160176   ERROR: PLEX.utils: _____________________________________________________________
19:36:52.210 T:4091846672   DEBUG: ------ Window Init (DialogNotification.xml) ------
19:36:52.213 T:3414160176   ERROR: Exception in thread Thread-3:
                                            Traceback (most recent call last):
                                              File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 173, in run
                                                self._run_internal()
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 253, in _run_internal
                                                self.start_library_sync(block=True)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 128, in start_library_sync
                                                library_sync.start(show_dialog, repair, self.on_library_scan_finished)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/full_sync.py", line 426, in start
                                                FullSync(repair, callback, show_dialog).run()
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/utils.py", line 1075, in wrapper
                                                result = func(*args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/full_sync.py", line 403, in run
                                                if not self.full_library_sync():
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/full_sync.py", line 380, in full_library_sync
                                                ctx.remove(plex_id, plex_type)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/itemtypes/movies.py", line 246, in remove
                                                self.kodidb.delete_artwork(kodi_id, kodi_type)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/kodi_db/common.py", line 167, in delete_artwork
                                                self.delete_cached_artwork(row[0])
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/kodi_db/common.py", line 34, in wrapper
                                                return method(self, *args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/kodi_db/common.py", line 172, in delete_cached_artwork
                                                self.artcursor.execute("SELECT cachedurl FROM texture WHERE url = ? LIMIT 1",
                                            AttributeError: 'NoneType' object has no attribute 'execute'
```
I doubt it's related but just thought i'd post it

**Comment by dazedcrazy** [comment]
Library shared. 

> 
> The other issue stems from the fact that you deactivated artwork sync. I'll fix it in the next version
I reset PKC options when i resetting the db by mistake and only noticed after a full library sync happened then turned artwork sync off

**Comment by croneter** [comment]
Powerful bug-hunting skill you have in any case 😉

**Comment by croneter** [comment]
Ehhhhhm - I cannot reproduce your isse. It works here 🙄 
```
NOTICE: PLEX.music: ADD album plex_id: 56191 - Name: A Day at the Races
NOTICE: PLEX.music: ADD album plex_id: 52840 - Name: Greatest Hits II
```
Not a single element get's skipped during sync here. That's really weird...

Could you please try to again reset the Kodi database in the PKC settings under Advanced? Sorry for that...  😞 . Do you have the chance to test using another client? Will the same two elements be skipped like you describe?

**Comment by dazedcrazy** [comment]
User 1: [kodi_1.log](https://github.com/croneter/PlexKodiConnect/files/2800106/kodi_1.log)
User 2: [kodi_2.log](https://github.com/croneter/PlexKodiConnect/files/2800107/kodi_2.log)
Both get ``Item not yet synced`` but on different albums

**Comment by croneter** [comment]
Nope, I cannot reproduce with my user and your library using LibreElec and Kodi 17.6 🙄

**Comment by dazedcrazy** [comment]
I thought i was being clever and i removed the Queen folder from my music library > ``Scan Library Files`` > ``Empty Trash`` > Added the folder back > ``Scan Library Files``. It's now a different artist and different albums/tracks missing.

**Comment by croneter** [comment]
I don't understand why I cannot reproduce your issue with exactly the same library and same PKC version 🙄

**Comment by croneter** [comment]
If you feel adventurous, I can give you some instructions on how to download the PMS metadata just like PKC. We'd see whether you could reproduce the issue "outside" of PKC (I'm starting to suspect that it's a Plex thing...)

**Comment by dazedcrazy** [comment]
Yeah just let me know what to do.
My log files does have a few different errors after being idle all day maybe there is something there.

This one is like every 2 minutes in the log not a debug log
```
ERROR: PLEX.plexgdm: Could not download clients for 192-168-0-4.08f9019b085b4071bbb7a74e56a22399.plex.direct
```
Then this one
```xml
12:57:36.959 T:3380855600   ERROR: PLEX.utils: Error encountered: sync.py crashed - u'kodi_fileid'
12:57:36.965 T:3380855600   ERROR: PLEX.utils: _____________________________________________________________
12:57:36.966 T:3380855600   ERROR: PLEX.utils:     Traceback (most recent call last):
12:57:36.966 T:3380855600   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 173, in run
12:57:36.966 T:3380855600   ERROR: PLEX.utils:         self._run_internal()
12:57:36.967 T:3380855600   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 303, in _run_internal
12:57:36.967 T:3380855600   ERROR: PLEX.utils:         library_sync.store_websocket_message(message)
12:57:36.968 T:3380855600   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/websocket.py", line 44, in store_websocket_message
12:57:36.968 T:3380855600   ERROR: PLEX.utils:         process_playing(message['PlaySessionStateNotification'])
12:57:36.968 T:3380855600   ERROR: PLEX.utils:       File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/websocket.py", line 283, in process_playing
12:57:36.969 T:3380855600   ERROR: PLEX.utils:         PLAYSTATE_SESSIONS[session_key]['file_id'] = typus['kodi_fileid']
12:57:36.969 T:3380855600   ERROR: PLEX.utils:     KeyError: u'kodi_fileid'
12:57:36.969 T:3380855600   ERROR: PLEX.utils: _____________________________________________________________
12:57:36.981 T:3380855600   ERROR: Exception in thread Thread-3:
                                            Traceback (most recent call last):
                                              File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 173, in run
                                                self._run_internal()
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/sync.py", line 303, in _run_internal
                                                library_sync.store_websocket_message(message)
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/websocket.py", line 44, in store_websocket_message
                                                process_playing(message['PlaySessionStateNotification'])
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/library_sync/websocket.py", line 283, in process_playing
                                                PLAYSTATE_SESSIONS[session_key]['file_id'] = typus['kodi_fileid']
                                            KeyError: u'kodi_fileid'
```

**Comment by croneter** [comment]
The entire music library?!?

**Comment by dazedcrazy** [comment]
Yeah it's empty on restart

**Comment by croneter** [comment]
Ouch

**Comment by dazedcrazy** [comment]
It's hard to tell but i think it just started happen with the 2.6.3 update

**Comment by ashleybone** [comment]
Looks like this has been fixed, but I wanted to confirm that the fix to #662 that's in 2.6.3 is why your entire music library is now getting removed.  I started playing around with importing my music library last night and saw the same problem.  With 2.6.2, only a single track was deleted each sync due to the issue in #662.

**Comment by dazedcrazy** [comment]
Tested https://github.com/croneter/PlexKodiConnect/pull/690 after multiple restarts everything still seems to be there. Great job

**Comment by croneter** [comment]
Should be solved with stable and beta 2.6.4 - both issues 😉 . Let me know here if this is not the case

**Comment by ashleybone** [comment]
Everything is working perfectly for me now.
